dallinger/recruiters.py

```diff
@@ -381,9 +381,9 @@ def normalize_entry_information(self, entry_information: dict):
         """Map Prolific Study URL params to our internal keys."""
 
         participant_data = {
-            "hit_id": entry_information["STUDY_ID"],
-            "worker_id": entry_information["PROLIFIC_PID"],
-            "assignment_id": entry_information["SESSION_ID"],
+            "hit_id": entry_information.get("STUDY_ID", generate_random_id()),
+            "worker_id": entry_information.get("PROLIFIC_PID", generate_random_id()),
+            "assignment_id": entry_information.get("SESSION_ID", generate_random_id()),
```

Comment:
I think this is dangerous in some situations; in particular, it is useful that if someone forgets the URL parameters an error is thrown. I think `generate_random_id` should only be called if `generate_tokens=True`. Some logic like this is already present in `prepare_advertisement`:

```py
def prepare_advertisement():
    session = db.session
    config = _config()
    mode = config.get("mode")

    # Browser rule validation, if configured:
    browser = ValidatesBrowser(config)
    if not browser.is_supported(request.user_agent.string):
        raise ExperimentError("browser_type_not_allowed")

    entry_information = request.args.to_dict()

    if entry_information.get("generate_tokens", None) in ("1", "true", "yes"):
        redirect_params = entry_information.copy()
        del redirect_params["generate_tokens"]
        for entry_param in ("hitId", "assignmentId", "workerId"):
            if not redirect_params.get(entry_param):
                redirect_params[entry_param] = generate_random_id()
        return True, {"redirect": redirect(url_for("advertisement", **redirect_params))}
```

Shouldn't this logic be sufficient? Is it not working?